# 기계학습 (60점)

### 가공식품의 영양성분표 데이터
dataurl = https://raw.githubusercontent.com/Datamanim/datarepo/main/adp/mock/003_m1.csv    
출처 : https://www.data.go.kr/data/15047698/fileData.do 후처리     

- 식품명 : {카테고리}_제품명 순서로 작성된다. unique한 카테고리는 5개 (라면,시리얼,식빵,아이스밀크,잼)이다. 
    - 식품명은 군집분석 모델링을 할 때 사용하지 않는다.
- 업체명 : 업체명
- 그 외 : 해당 컬럼명의 정량적인 값

In [27]:
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/adp/mock/003_m1.csv')
df.head()

식품명  에너지(kcal) 영양성분함량기준량  수분(g)  단백질(g)  지방(g)  회분(g)  탄수화물(g)  \
0    라면_스낵면Cup        427      100g    0.0    8.06  14.52    0.0    66.13   
1      라면_준코라면        417      100g    0.0    8.33  12.50    0.0    67.50   
2    라면_오뚜기김치면        429      100g    0.0    8.57  13.33    0.0    68.57   
3  라면_오뚜기 마열라면        438      100g    0.0   10.48  15.24    0.0    64.76   
4     라면_컵삼양라면        454      100g    0.0    7.69  18.46    0.0    64.62   

   당류(g)  식이섬유(g)  ...  베타카로틴(μg)  티아민(mg)  리보플라빈(mg)  니아신(mg)  비타민 C(mg)  \
0   4.84      0.0  ...        0.0      0.0        0.0      0.0        0.0   
1   4.17      0.0  ...        0.0      0.0        0.0      0.0        0.0   
2   3.81      0.0  ...        0.0      0.0        0.0      0.0        0.0   
3   2.86      0.0  ...        0.0      0.0        0.0      0.0        0.0   
4   3.08      0.0  ...        0.0      0.0        0.0      0.0        0.0   

   비타민 D(μg)  콜레스테롤(mg)  포화지방산(g)  트랜스지방산(g)                 업체명  
0        0.0        0.0      5.48        0.0  오뚜기라면(주),해당없음,해당없음  
1        0.0        0.0      6.67        0.0     (주)팔도,해당없음,해당없음  
2        0.0        0.0      6.67        0.0  오뚜기라면(주),해당없음,해당없음  
3        0.0        0.0      7.62        0.0  오뚜기라면(주),해당없음,해당없음  
4        0.0        0.0      9.23        0.0   삼양식품(주),해당없음,해당없음  

[5 rows x 27 columns]

### 기계학습1    
시각화 및 EDA

### 기계학습2
식품명을 모르는 상태에서 영양성분표를 바탕으로 데이터를 군집화 작업을 하려한다.      
군집 모델링 전 차원 축소가 필요한지 확인하고 필요하다면 그 근거를 작성하고 차원축소를 진행하라.      

### 기계학습3      
위에서 전처리한 데이터를 기준으로 Kmeans 군집을 실시하라. 최적의 군집 수를 도출한 근거를 설명하라.         
해당 군집 결과를 PCA를 진행한 두개의 축위에 각 클러스터별 다른 색상으로 산점도를 그려라.

### 기계학습4
2에서 전처리한 데이터를 3에서 추출한 최적군집 숫자로 계층적 군집분석을 실시하라.    
모델링시 군집간 거리 4가지 방법(최소값,최대값,평균간 거리, 평균)으로 다르게 하여 군집분석을 실시하고    
3에서 추출한 PCA 축상에 군집별 산점도를 출력하라

### 기계학습5
3,4에서 도출한 5개의 군집 결과 그래프와 실제 식품명의 카테고리를 비교 분석하여 이전 군집 분석에 대한 결과를 정리하라     

### 서울특별시 전월세가
dataurl = https://raw.githubusercontent.com/Datamanim/datarepo/main/adp/mock/003_m2.csv    
출처 : https://data.seoul.go.kr/dataList/OA-21276/S/1/datasetView.do 후처리    
        
23년도 강서구의 전월세 실거래가 데이터이다.    
데이터를 바탕으로 총 소모비용 ( 보증금 + 임대료 * 계약기간(월))을 예측하는 모델을 만들려고 한다.

In [2]:
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/adp/mock/003_m2.csv')

In [3]:
df

자치구명 법정동명       지번   층       계약일 전월세구분  보증금(만원)  임대료(만원)    건축년도   건물용도  \
0      강서구  화곡동   343-66   2  20230102    월세     1000       60  2004.0   오피스텔   
1      강서구  화곡동   343-66   2  20230211    월세     1000       60  2004.0   오피스텔   
2      강서구  화곡동    402-4   4  20230208    월세     2000       55  2000.0   오피스텔   
3      강서구  가양동   1479-2   6  20230702    월세      500       42  2005.0   오피스텔   
4      강서구  화곡동   343-66   2  20230401    월세      500       60  2004.0   오피스텔   
...    ...  ...      ...  ..       ...   ...      ...      ...     ...    ...   
25525  강서구  화곡동   424-23   4  20231207    전세    16000        0  2016.0  연립다세대   
25526  강서구  화곡동  105-357   5  20231207    전세    21500        0  2018.0  연립다세대   
25527  강서구  화곡동   1091-0  11  20231207    전세    61000        0  2002.0    아파트   
25528  강서구  화곡동  410-296   4  20231207    전세    12500        0  2001.0  연립다세대   
25529  강서구  마곡동   797-14  14  20231207    월세    16870        8  2016.0   오피스텔   

              계약기간  임대평수  
0      23.01~25.01   8.1  
1      23.02~24.02   8.4  
2      23.03~25.03  11.5  
3      23.07~24.07   6.1  
4      23.05~24.05   8.3  
...            ...   ...  
25525  24.02~26.02   9.6  
25526  23.12~25.12   7.4  
25527  23.12~25.12  25.7  
25528  24.01~26.01  14.1  
25529  24.01~26.01   7.3  

[25530 rows x 12 columns]

### 기계학습5
시각화를 포함한 EDA

### 기계학습6
모델링을 위한 데이터 전처리를 시행하라. 추가로 건물용도 마다 각각 여러 수치형변수(임대평수, 건물 건축 기간(2024년기준), 소모비용)간의 통계적 차이가 있는지 확인하라    

### 기계학습7
xgb,lightgbm,randomforest 각 모델을 학습하고 각 모델별로 주요 변수들을 확인하고 해석하라

# 통계 (40점)

### 통계1     
학교내 학생들의 시험 점수를 나타낸 것이다. 성별과 학년이 점수에 대해 가지는 주효과 교호작용효과를 알아보고자 한다.    
(1) 주효과와 교호작용 효과에 대한 가설 설정     
(2) 통계검정을 사용하기 위한 사전 조건 확인    
(3) 통계검정
(4) 사후분석

In [7]:
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/adp/mock/003_s4.csv')
df.head()

성별   학년  점수
0  남학생  1학년  70
1  남학생  1학년  76
2  남학생  1학년  78
3  남학생  1학년  67
4  남학생  1학년  71

### 통계2      
일자별 장마감시간의 주식의 상승과 하락을 추적한 데이터이다.     
주가 변동이 무작위 인지 특정 경향성을 가지는지 통계적으로 확인하려 한다.       

(1) 가설설정     
(2) 검정 통계량 확인 및 분석     

dataurl = https://raw.githubusercontent.com/Datamanim/datarepo/main/adp/mock/003_s2.csv

In [2]:
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/adp/mock/003_s2.csv')
df.head()

Price Change
0           상승
1           하락
2           상승
3           상승
4           하락

### 통계3    
연구자가 새로운 수면 개선 약을 개발했으며, 이 약의 효과를 검증하기 위해 10명의 피험자에게 일주일 동안 매일 밤 약을 복용하도록 하고, 복용 전과 복용 후의 평균 수면 시간을 비교했다. 연구자는 약이 수면 시간에 유의미한 변화를 가져다주었는지 확인하고 싶어 한다.      
(1) 가설 설정     
(2) 통계 검정

| 피험자 | 복용 전 평균 수면 시간(시간) | 복용 후 평균 수면 시간(시간) |
|---|------------------|------------------|
| 1 | 6 | 7 |
| 2 | 7 | 8 |
| 3 | 6.5 | 6.8 |
| 4 | 5 | 6.5 |
| 5 | 7 | 7.5 |
| 6 | 5.5 | 5 |
| 7 | 6.7 | 7 |
| 8 | 6.2 | 6.8 |
| 9 | 7 | 7.2 |
| 10 | 5.5 | 6 |

### 통계4
학생 20명을 무작위로 두 그룹으로 나누어 카페인이 학습에 효과가 있는지 확인하려한다. 두 그룹간의 평균 점수 차이가 존재하는지 확인하라    
카페인 그룹 테스트 점수: [88, 92, 94, 95, 96, 97, 91, 93, 98, 94]     
플라시보 그룹 테스트 점수: [81, 83, 84, 88, 86, 85, 87, 82, 80, 83]    
 
(1) 가설설정    
(2) 검정